frontend/src/forms/subforms/AddressForm.tsx

```diff
@@ -42,17 +42,12 @@ const AddressForm = <T extends any>(props: AddressProps & FormikProps<T>) => {
   };
 
   const postalCodeFormatter = (postal: string) => {
-    const noSpace = RegExp(/^[a-zA-z][0-9][a-zA-z][0-9][a-zA-z][0-9]$/);
-    const hyphen = RegExp(/^[a-zA-z][0-9][a-zA-z]-[0-9][a-zA-z][0-9]$/);
-    if (noSpace.test(postal)) {
-      postal = postal.substr(0, 3) + ' ' + postal.substr(3, 5);
-      return postal.toUpperCase();
-    } else if (hyphen.test(postal)) {
-      postal = postal.replace('-', ' ');
-      return postal.toUpperCase();
-    } else {
-      return postal.toUpperCase();
+    const regex = /[a-zA-z][0-9][a-zA-z][\s-]?[0-9][a-zA-z][0-9]/;
+    const format = postal.match(regex);
+    if (format !== null) {
```

Comment:
this works, and I don't think we need to rework this again, but what I meant was something like (pseudocode)
const regex = /([a-zA-z][0-9][a-zA-z])[\s-]?*([0-9][a-zA-z][0-9])/;
const matches = postal.match(regex);
if(format !== null and format.length === 2) {
postal = `${format[0]} ${format[1]}`
}

Reply:
also making comments with backticks in github is a pain.

Reply:
Ohh yes that is way better